scipy/ndimage/_multimethods.py

```diff
@@ -0,0 +1,1065 @@
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_zoom_arg_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def zoom(input, zoom, output=None, order=3, mode='constant', cval=0.0,
+         prefilter=True, *, grid_mode=False):
+    return input, _mark_non_coercible(output)
+
+
+def _rotate_arg_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for ``rotate``.
+    """
+    def self_method(input, angle, axes=(1, 0), reshape=True,
+                    output=None, *args, **kwargs):
+        return (dispatchables[0], angle, axes, reshape,
+                dispatchables[1]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_rotate_arg_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def rotate(input, angle, axes=(1, 0), reshape=True, output=None, order=3,
+           mode='constant', cval=0.0, prefilter=True):
+    return input, _mark_non_coercible(output)
+
+
+############################################
+# measurement multimethods
+############################################
+
+
+def _input_structure_output_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for signatures involving required
+    ``input`` with ``structure`` and ``output`` kwargs.
+    """
+    def self_method(input, structure=None, output=None, *args, **kwargs):
+        return (dispatchables[0], dispatchables[1],
+                dispatchables[2]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_input_structure_output_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def label(input, structure=None, output=None):
+    return input, structure, _mark_non_coercible(output)
+
+
+def _input_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for signatures involving
+    the required ``input`` as the only dispatchable.
+    """
+    def self_method(input, *args, **kwargs):
+        return (dispatchables[0],) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_input_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def find_objects(input, max_label=0):
+    return (input,)
+
+
+def _input_labels_index_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for signatures involving required
+    ``input`` with ``labels`` and ``index`` kwargs.
+    """
+    def self_method(input, labels=None, index=None, *args, **kwargs):
+        return (dispatchables[0], dispatchables[1],
+                dispatchables[2]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def labeled_comprehension(input, labels, index, func, out_dtype,
+                          default, pass_positions=False):
+    return input, labels, index
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def sum_labels(input, labels=None, index=None):
```

Comment:
The index argument here and on most or all other methods in `_measurements.py` needs to also accept an integer, rather than always being coerced to an array.

Reply:
I'm marking them as `_mark_index` and finally handling those also in `__uarray_convert__`. Does that sound right?